validators/types.go

```diff
@@ -0,0 +1,65 @@
+package validators
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+var (
+	ErrMismatchValidatorType    = errors.New("mismatch between validator and validator set")
+	ErrMismatchValidatorSetType = errors.New("mismatch between validator sets")
+	ErrValidatorAlreadyExists   = errors.New("validator already exists in validator set")
+	ErrValidatorNotFound        = errors.New("validator not found in validator set")
+)
+
+type ValidatorType string
+
+const (
+	ECDSAValidatorType ValidatorType = "ecdsa"
+	BLSValidatorType   ValidatorType = "bls"
+)
+
+func (t *ValidatorType) FromString(s string) error {
+	x := ValidatorType(s)
+
+	switch x {
+	case ECDSAValidatorType, BLSValidatorType:
+		*t = x
+
+		return nil
+	default:
+		return fmt.Errorf("invalid validator type: %s", s)
+	}
+}
+
+// Validator defines the interface of the methods a validator implements
+type Validator interface {
+	Type() ValidatorType
+	String() string
+	Addr() types.Address
+	Copy() Validator
+	Equal(Validator) bool
+	MarshalRLPWith(*fastrlp.Arena) *fastrlp.Value
+	UnmarshalRLPFrom(*fastrlp.Parser, *fastrlp.Value) error
+	Bytes() []byte
+	SetFromBytes([]byte) error
+}
+
+// ValidatorSet defines the interface of the methods validator set implements
+type Validators interface {
```

Comment:
Mismatch between the comment and type name

Reply:
Fixed at 07ff8c461deea0789fd784aba63f79a85e78549f

Reply:
You've introduced a new mismatch here:
```go
// ParseIBFTType converts a ibftType string representation to a IBFTType
func ParseValidatorType(validatorType string) (ValidatorType, error) {
```

Reply:
Fixed e9ca24d1c38b36baf063a95e249c44c87a7098e9